operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java

```diff
@@ -217,14 +218,20 @@ public NextAction onSuccess(Packet packet, CallResponse<T> callResponse) {
         if (isKnownFailedJob(job)) {
           return doNext(cleanUpAndReintrospect(), packet);
         } else if (job != null) {
-          return doNext(processIntrospectionResults(), packet);
+          return doNext(processIntrospectionResults(), packet).withDebugComment(job, this::foundJobDescription);
```

Comment:
The logic has changed in PR 2609 to account for failures due to image pull error and introspector job timeout.

Reply:
Understood.